cmd/autothrottle/internal/api/api_test.go

```diff
@@ -9,7 +9,18 @@ import (
 	"github.com/DataDog/kafka-kit/v4/kafkazk"
 )
 
+var (
+	trigger = make(chan struct{}, 10)
+)
+
+func clearTrigger() {
+	for len(trigger) > 0 {
```

Comment:
nit: Generally using `len(<channel>)` is discouraged, because if someone else reads from the channel before we execute L18 then we have blocked this thread, most likely unintentionally. That race condition wouldn't occur in this code base, but I'd still recommend clearing the channel with something like
```
for {
  select {
  case <-trigger:
  default:
  }
}
```
just to keep the code with the best practice.